### PR TITLE
Fixes Maruck bugs

### LINKED ISSUE #192: Red Hot Armor + Maruck the Marked
When active player plays Red Hot Armor and the opponent has Maruck out, he is capturing aember for the armor which is "removed" by RHA. By the way RHA is worded it seems this isn't as intended but cannot find a rules reference online regarding it. 

Red Hot Armor: "Play: Each enemy creature with armor loses all of its armor until the end of the turn and is dealt 1D for each point of armor it lost this way."

Maruck the Marked: "After Maruck the Marked prevents damage with its armor, capture 1A for each damage just prevented."

### LINKED ISSUE #412: UI: Maruck and Shield of Justice
Maruck is being protected by Shield of Justice, but its armor is being reduced (even though it is not taking the damage). The armor should not be reduced.

Screenshot of Maruck with Protect the Weak, fighting Shooler after Shield of Justice is played:

image

### PR DESCRIPTION
**Pull request by romanogit**
Fixes #192 Redhot armor should not allow Maruck to capture amber
Fixes #412 Dealing damage when it cannot affect a target should not reduce armor

### CODE REVIEW

File: server/game/GameActions/DealDamageAction.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ class DealDamageAction extends CardGameAction {
     setDefaultProperties() {
         this.amount = null;
         this.amountForCard = (card, context) => 1; // eslint-disable-line no-unused-vars
+        this.reduceArmorBy = (card, context) => 0; // eslint-disable-line no-unused-vars
```
Comment (cryogen) [inline server/game/GameActions/DealDamageAction.js]: the eslint disabling is not needed here

Comment (romanogit) [reply server/game/GameActions/DealDamageAction.js]: When I compiled, I got the context not used error.

Comment (cryogen) [reply server/game/GameActions/DealDamageAction.js]: Ah yes, just remove the variables from both, they're not needed

Comment (romanogit) [reply server/game/GameActions/DealDamageAction.js]: ok, I will make the change

Comment (romanogit) [reply server/game/GameActions/DealDamageAction.js]: I get the same error on card:
C:\workspace_svn\public\keyteki\br-bugs\server\game\GameActions\DealDamageAction.js
  6:31  error  'card' is defined but never used  no-unused-vars
  7:31  error  'card' is defined but never used  no-unused-vars

**Pull request merged**